cumulus/client/consensus/aura/src/collators/lookahead.rs

```diff
@@ -348,144 +348,135 @@ where
 
 			// Build in a loop until not allowed. Note that the authorities can change
 			// at any block, so we need to re-claim our slot every time.
-			let mut parent_hash = initial_parent.hash;
-			let mut parent_header = initial_parent.header;
+			let parent_hash = initial_parent.hash;
+			let parent_header = initial_parent.header;
 			let overseer_handle = &mut params.overseer_handle;
 
 			// Do not try to build upon an unknown, pruned or bad block
 			if !collator.collator_service().check_block_status(parent_hash, &parent_header) {
 				continue
 			}
 
-			// This needs to change to support elastic scaling, but for continuously
-			// scheduled chains this ensures that the backlog will grow steadily.
-			for n_built in 0..2 {
```

Comment:
What we actually want is that we in total are only always produce two blocks on top of the last backed block. What currently can happen if we missed to back a block, we put another block (assuming unincluded segment len of 3) on top. So, if everything works as expected, we will need the unincluded segment length of 3, but if a block failed to be backed we should go back to 2 from the included block.